app/decorators/module_overview_decorator.rb

```diff
@@ -13,18 +13,17 @@ def call_to_action
     end
   end
 
-  # @return [Array<String, Symbol, Array>]
-  def topics_by_submodule
-    mod.items_by_submodule.map do |num, items|
+  # @return [Array<Hash{Symbol => String, Symbol, Array}>]
+  def sections
+    mod.items_by_submodule.each.with_index(1).map do |(num, items), position|
       intro = items.first
-      content = items[1..]
-      [
-        num,                                  # submodule digit
-        intro.model.heading,                  # submodule intro heading
-        previous_submodule_heading_text(num), # previous submodule heading
-        status(content),                      # symbol - all content pages in the submodule
-        topics(submodule: num, items: items), # Array(String, Symbol)
-      ]
+      heading = num.nil? ? 'Module introduction' : intro.model.heading
```

Comment:
We can tidy this if we remove the comments and variables, setting `heading:` inline.